DOC: clarify .dsk and bmaptool use for EFI platforms only

dbkinder opened the pull request:
Issue #113 notes user tried using .dsk format and bmaptool
to bitbake an image and create a bootable USB stick for an
Intel Edison board.

This patch updates documentation to clarify
.dsk format and use of bmaptool is for EFI platforms only
and won't work for Intel Edison (or BeagleBone Black).

Also addressed some formatting errors and clarity edits.

[skip ci]

Signed-off-by: David Kinder <david.b.kinder@intel.com>

kad commented on doc/howtos/building-images.rst:
```diff
@@ -195,7 +204,7 @@ compressed formats:
 ``OSTRO_VM_IMAGE_TYPES``, set to any combination of the following::
 
-    dsk dsk.xz dsk.vdi dsk.vdi.xz
+    dsk dsk.xz dsk.zip dsk.vdi dsk.vdi.xz dsk.vdi.zip
```
recent addition of dsk.ova is missing (virtualbox appliance, replaced dsk.vdi*

gvancuts replied:
Could you also update local.conf.sample to reflect the very same changes?

gvancuts replied:
On a slightly different note, I noticed earlier it's possible to put in something like "dsk.vdi.bmap" in local.conf and such *.dsk.vdi.bmap file gets created (but no corresponding dsk.vdi)... should this be filed as a bug?

gvancuts replied:
Final thought on this one, we should describe how people can create images that can be flashed with bmaptool (and I mean here that a .bmap file would be produced along with the image). @kad what is the right approach? Would putting something like OSTRO_VM_IMAGE_TYPES = "dsk.xz.bmap" do the trick?

kad replied:
dsk.xz.bmap doesn't make sense. bmap file is generated on original, non-compressed sparse image file. Practically all compression methods are killing sparseness.

dsk.bmap can be used with multiple compressions, it applies both to dsk.xz and with development version of bmaptool also to dsk.zip.

As for dsk.vdi.bmap, it's expected that dsk.vdi not present, if dsk.vdi not explicitly in mentioned in types. Intermediate compression results are not preserved if user didn't ask for them.

And in overall, with arrival of OVA, vdi type should go away.

kad commented:
Looks good, just update OVA format for VirtualBox, dsk.vdi is obsoleted nowadays with real OVA appliance format.

gvancuts commented on doc/howtos/building-images.rst:
```diff
@@ -368,22 +381,24 @@ by adding these lines to your ``local.conf`` file::
-slightly differently than layers and have to be used here because
-layer names are not available internally) it is expected to come::
+The file ``conf/my-supported-recipes.txt`` is created to
+specifie the recipe and which "collection" it is expected to come from::
```
s/specifie/specify

dbkinder replied:
fixed

dbkinder commented:
removed references to .vdi images (using .ova now instead) both in building-images and booting-and-installation docs.  I did not update local.conf.sample thinking that's a dev team thing to modify @gvancuts @pohly

pohly commented:
It would be fine and (in fact) very welcome if you could also update local.conf.sample. We are all in this together :-)

dbkinder commented:
OK.  I'll do that in a separate PR since it's outside the /doc folder

pohly commented:
This contains some unrelated doc cleanup. Was that intentional? No blocker for merging, just not good practice.

@kad okay to merge now?

dbkinder commented:
Sorry about that.  I found some formatting improvements while I was in the files, and addressed the comments with some changes unrelated to the original intent of this PR. I'll do them in separate PRs next time.

kad commented:
ok for me.
